pkg/service/localstore.go

```diff
@@ -50,13 +50,15 @@ func (s *LocalStore) LoadRoom(_ context.Context, name livekit.RoomName) (*liveki
 	return room, nil
 }
 
-func (s *LocalStore) ListRooms(_ context.Context, names []livekit.RoomName) ([]*livekit.Room, error) {
+func (s *LocalStore) ListRooms(_ context.Context, names []livekit.RoomName, sids []livekit.RoomID) ([]*livekit.Room, error) {
 	s.lock.RLock()
 	defer s.lock.RUnlock()
 	rooms := make([]*livekit.Room, 0, len(s.rooms))
 	for _, r := range s.rooms {
-		if names == nil || funk.Contains(names, livekit.RoomName(r.Name)) {
-			rooms = append(rooms, r)
+		if (names == nil && sids == nil) || funk.Contains(names, livekit.RoomName(r.Name)) || funk.Contains(sids, livekit.RoomID(r.Sid)) {
+			if !funk.Contains(rooms, r) {
```

Comment:
don't think this is needed.

Reply:
Same as above. Have to check for `sid` match. Wonder how the overlapping test passed (where `names` and `sids` point to the same room). It returned only one object. Have to check that.

Reply:
Okay, I am mistaken. It is looping over all rooms and checking for matches. So, yeah, the contains check is not needed. Will remove